camel-core/src/main/java/org/apache/camel/ComponentVerifier.java

```diff
@@ -21,118 +21,287 @@
+         */
         CONNECTIVITY;
 
         private static final Scope[] VALUES = values();
-
+        /**
+         * Get an instance of this scope from a string representation
+         *
+         * @param scope the scope as string, which can be in any case
+         * @return the scope enum represented by this string
+         */
         public static Scope fromString(String scope) {
-            for (int i = 0; i < VALUES.length; i++) {
-                if (ObjectHelper.equal(scope, VALUES[i].name(), true)) {
-                    return VALUES[i];
+            for (Scope value : VALUES) {
+                if (ObjectHelper.equal(scope, value.name(), true)) {
+                    return value;
                 }
             }
-
             throw new IllegalArgumentException("Unknown scope <" + scope + ">");
         }
     }
 
+    // =============================================================================================
+
     /**
-     * Represent an error
+     * This interface represents a detailed error in case when the verification fails.
      */
-    interface Error extends Serializable {
+    interface VerificationError extends Serializable {
+
         /**
-         * @return the error code
+         * The overall error code, which can be either a {@link StandardCode} or a custom code. It is
+         * recommended to stick to the predefined standard codes
+         *
+         * @return the general error code.
          */
-        String getCode();
+        Code getCode();
 
         /**
+         * A human readable description of the error in plain english
+         *
          * @return the error description (if available)
          */
         String getDescription();
 
         /**
-         * @return the parameters in error
+         * A set of input parameter names which fails the verification. These are keys to the parameter provided
+         * to {@link #verify(Scope, Map)}.
+         *
+         * @return the parameter names which are malformed and caused the failure of the validation
          */
-        Set<String> getParameters();
+        Set<String> getParameterKeys();
 
         /**
-         * @return a number of key/value pair with additional information related to the validation.
+         * Details about the failed verification. The keys can be either predefined values
+         * ({@link ExceptionAttribute}, {@link HttpAttribute}, {@link GroupAttribute}) or it can be free-form
+         * custom keys specific to a component. The standard attributes are defined as enums in all uppercase (with
+         * underscore as separator), custom attributes are supposed to be in all lower case (also with underscores
+         * as separators)
+         *
+         * @return a number of key/value pair with additional information related to the verification.
          */
-        Map<String, Object> getAttributes();
-    }
+        Map<Attribute, Object> getDetails();
```

Comment:
It may be nice to have something like:

```java
default Object getDetail(String name) {
    Map<Attribute, Object> details = getDetails();
    if (details != null) {
        for (Map.Entry<Attribute, Object> entry : details.entrySet()) {
            if (ObjectHelper.equals(entry.getKey(), name)) {
                return entry.getValue();
            }
        }

        return null;
    }
}